src/components/datatable.component.ts

```diff
@@ -790,6 +790,9 @@ export class DatatableComponent implements OnInit, AfterViewInit, DoCheck {
   recalculatePages(): void {
     this.pageSize = this.calcPageSize();
     this.rowCount = this.calcRowCount();
+    while(this.rowCount / this.pageSize < this.offset) { 
```

Comment:
suppose `rowCount` is `15`, `pageSize` is `10` that means table can have at max 2 pages(0 and 1 `offset` value) 10 rows in first and 5 rows in second). This was checking `while` current `offset` is greater than 1, then decrement it.
BTW, I have replaced the while loop with math function in https://github.com/swimlane/ngx-datatable/pull/549/commits/00eb63af1848b2635babd64a3440c5b58db2dcbd which directly checks if current `offset` is greater than number of required pages in table, then set `offset` to maximum possible page:
```
if(this.rowCount / this.pageSize < this.offset) {
       this.offset = Math.floor(this.rowCount / this.pageSize);
}
```
Here in example we took, suppose there are 15 available rows and limit is 10 then offset will be set to Math.min(15/10) => Math.min(1.5) => 1